save-agent/src/linuxX64Main/kotlin/com/saveourtool/save/agent/SaveAgent.kt

```diff
@@ -121,6 +141,57 @@ class SaveAgent(private val config: AgentConfiguration,
         coroutineScope.cancel()
     }
 
+    // a temporary workaround for python integration
+    private fun executeAdditionallySetup(targetDirectory: Path, additionalFiles: List<FileKey>) = runCatching {
+        additionalFiles
+            .singleOrNull { it.name == "setup.sh" }
+            ?.let { fileKey ->
+                val targetFile = targetDirectory / fileKey.name
+                logDebugCustom("Additionally setup of evaluated tool by $targetFile")
+                val setupResult = ProcessBuilder(true, FileSystem.SYSTEM)
+                    .exec(
+                        targetFile.toString(),
+                        "",
+                        null,
+                        SETUP_SH_TIMEOUT
+                    )
+                if (setupResult.code != 0) {
+                    throw IllegalStateException("${fileKey.name} is failed with error: ${setupResult.stderr}")
+                }
+            }
+    }
+
+    // prepare save-overrides.toml based on config.save.*
+    private fun prepareSaveOverridesToml(targetDirectory: Path) = runCatching {
+        with(config.save) {
+            val generalConfig = buildMap {
```

Comment:
Waiting for ktoml to support serialization? 🤔

Reply:
actually, I wanted to construct `WarnPluginConfig` directly and then write it to a toml file. But it requires `regex` library for linuxx64. (similar as for 'coroutines')

Reply:
Isn't regex support completely provided by Kotlin standard library?

Reply:
looks like the issue in IDE:
![image](https://user-images.githubusercontent.com/1164369/187628961-88ec2bf9-d68b-439b-a307-15732f36e973.png)

Reply:
I think there is no dependency on `warn-plugin` in save-agent

Reply:
problem with `regex` there.

The following change fixes it
![image](https://user-images.githubusercontent.com/1164369/187637375-768fbf6e-7ab4-4881-9672-eb36e1779ed2.png)

@petertrr @akuleshov7 is it okay to apply this?

Reply:
It looks really strange. If it helps simplifying code, then maybe, but otherwise I'd rather not.

Reply:
will try with using Toml's serialization then

Reply:
KToml doesn't support serialization. Will create a separate task to add more targets for save-agent to simplify debugging